db-service/lib/cqn4sql.js

```diff
@@ -1828,7 +1828,12 @@ function cqn4sql(originalQuery, model = cds.context?.model || cds.model) {
    * @returns true if the given definition shall be localized
    */
   function isLocalized(definition) {
-    return inferred.SELECT?.localized && definition['@cds.localized'] !== false
+    return (
+      inferred.SELECT?.localized &&
+      definition['@cds.localized'] !== false &&
+      !inferred.SELECT.forUpdate &&
+      !inferred.SELECT.forShareLock
```

Comment:
Does that mean that we receive queries like that, with `localized` and `forUpdate` set?: 
```js
SELECT.localized(Foo).foUpdate()
```
→ looks like a glitch in the caller, and should be fixed there?
At least as well

Reply:
This was triggered by `cap/issues/issues/14177` which was facing exactly this scenario. I don't know the exact details, but should probably be followed up with the reporter.